src/mlpack/core/optimizers/sgdr/snapshot_sgdr_impl.hpp

```diff
@@ -64,16 +61,17 @@ double SnapshotSGDR<DecomposableFunctionType, UpdatePolicyType>::Optimize(
     optimizer.DecayPolicy().StepSize() = optimizer.StepSize();
   }
 
+  optimizer.DecayPolicy().EpochBatches() = function.NumFunctions() /
+      double(batchSize);
+
   // If a user changed the batch size we have to update the restart fraction
   // of the cyclical decay instantiation.
   if (optimizer.BatchSize() != batchSize)
   {
     batchSize = optimizer.BatchSize();
-    optimizer.DecayPolicy().EpochBatches() = function.NumFunctions() /
-        double(batchSize);
```

Comment:
See the change of `src/mlpack/core/optimizers/sgdr/cyclical_decay.hpp`. In the past we compute `epochBatches` at CTOR, but now we cannot do this as we won't pass in `function` to CTOR anymore. So I just compute this value every time before calling the `Optimize`